bin/install-qa-check.d/60pkgconfig

```diff
@@ -1,12 +1,70 @@
 # Check for pkg-config file issues
 
 pkgconfig_check() {
+	# Make a list of .pc files and bail out if there aren't any
+	mapfile -d '' files < <(find "${ED}"/usr/lib*/pkgconfig "${ED}"/usr/share/pkgconfig -maxdepth 1 -type f -name '*.pc' -print0 2>/dev/null)
+	[[ -n "${files[@]}" ]] || return
+
+	local f
+
 	# Look for leaking LDFLAGS into pkg-config files
-	local f=$(egrep -sH '^Libs.*-Wl,(-O[012]|--hash-style)' "${ED}"/usr/*/pkgconfig/*.pc)
+	f=$(egrep -sH '^Libs.*-Wl,(-O[012]|--hash-style)' "${files[@]}")
 	if [[ -n ${f} ]] ; then
 		eqawarn "QA Notice: pkg-config files with wrong LDFLAGS detected:"
 		eqawarn "${f//${D}}"
 	fi
+
+	# Validate using pkgconfig
+	# Note: some less common implementations may not support this?
+	# Seems like f.d.o, OpenBSD, and of course pkgconf do though.
+	if ! pkg-config --validate "${files[@]}" ; then
+		eqawarn "QA Notice: pkg-config files which fail validation found!"
+		eqawarn "Run 'pkg-config --validate ...' for more information"
+	fi
+
+	# Check for unexpected paths
+	# e.g. https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=c90ab38e3577aae61fac2341b34ad593948de1cd
+	if [[ -n "${EPREFIX}" ]] ; then
+		for f in "${files[@]}" ; do
+			local key
+			for key in prefix exec_prefix libdir includedir ; do
+				local value=$(pkg-config --variable=${key} ${f})
+
+				if [[ ${value#${EPREFIX}} == ${value} ]] ; then
+					eqawarn "QA Notice: pkg-config files not respecting EPREFIX found"
+					eqawarn "${f}'s key=${key} does not respect EPREFIX:"
+					eqawarn "${key}=${value}"
+				fi
+			done
+		done
+	fi
+
+	# TODO: Generalise for non-lib64 libdir? Not that this is very common now
+	# that riscv chose a more standard layout.
+	#
+	# If we're installing to ${ED}/usr/lib/pkgconfig, let's make sure
+	# we're not referencing lib64.
+	#
+	# e.g. https://bugs.gentoo.org/729642
+	for f in "${files[@]}" ; do
+		# In ${ED}/usr/lib, we shouldn't reference lib64
+		if [[ ${f} == *lib/pkgconfig* ]] ; then
+			if [[ -d "${ED}"/usr/lib && ! -L "${ED}"/usr/lib ]] ; then
+				# (Don't bother if /usr/lib is a symlink to /usr/lib64)
+				continue
+			fi
+
+			if egrep -q "/lib64" ${f} ; then
+				eqawarn "QA Notice: pkg-config files not respecting libdir found"
+				eqawarn "${f} contains reference to 'lib64'"
+			fi
+		elif [[ ${f} == *lib64/pkgconfig* ]] ; then
+			if egrep -q "/lib[/]?$" ${f} ; then
```

Comment:
That certainly won't match inline `/lib/foo`.

Reply:
`grep -P '/lib(?![[:digit:]])'` maybe? yes sue me for using PCRE